src/memory/virtualMemory.ts

```diff
@@ -67,6 +70,30 @@ export class VirtualMemory {
         heapBaseAddress + heapSize * PageTable.ENTRY_SIZE
       )
     );
+    this.ebp = stackBaseAddress;
+    this.esp = stackBaseAddress - PageTable.ENTRY_SIZE;
+  }
+
+  public stackAllocate(data: number): number {
+    this.esp += PageTable.ENTRY_SIZE;
```

Comment:
Is it intended for the stack to grow from lower to higher addresses (instead of the other way around like in x86)?

Reply:
Yes because it was too hard to think the opposite way when I first did this :( But we can change it later without affecting anything outside the virtual memory.